gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/services/reflect/DatabaseClasses.java

```diff
@@ -314,6 +314,23 @@ public final Class loadApplicationClass(ObjectStreamClass classDescriptor)
 		return loadApplicationClass(classDescriptor.getName());
 	}
 
+	public final Class loadClassFromDB(String name) throws ClassNotFoundException {
+		ClassNotFoundException cnfe = new ClassNotFoundException(name);
+		if (applicationLoader == null) throw cnfe;
+		// if DDL replay has not started then ignore
+		final GemFireStore memStore = GemFireStore
+				.getBootingInstance();
+		if (memStore == null || !(memStore
+				.initialDDLReplayInProgress()
+				|| memStore.initialDDLReplayDone())) {
+			throw cnfe;
+		}
+		Class c = applicationLoader.loadClass(name, true);
+		if (c == null)
+			throw cnfe;
+		return c;
+
+	}
 	public boolean isApplicationClass(Class theClass) {
```

Comment:
What is the problem with the existing loadApplicationClass method? Even if there is some reason to add this, please see the initial block that checks for "com.pivotal.gemfirexd.internal." to prevent internal classes being overwritten. Also null check for applicationLoader.